### ISSUE TITLE
Interface regressions from 1.0

### ISSUE THREAD

**Issue by bgamari** [projecthamster/hamster#471]
Until very recently I had used `hamster-time-tracker-1.04`, which I was pretty happy with. Due to the removal of GConf from my distribution I have switched to 2.0 (or rather, `master`). Unfortunately, the switch hasn't been entirely pleasant; while one can always expect some amount of discomfort when user interfaces change, in this case the new interface really does feel like a regression in some respects:

 * [ ] the behavior of the "Add Activity" button is quite confusing, acting to either add a new activity or edit an existing activity depending upon whether there is an active selection
 * [ ] the collapsed "Overview" panel is significantly less useful than the previous Overview window as it is no longer possible to use the 
 * [ ] attempting to add an activity sometimes fails inexplicably and silently; that is, one clicks "Add Activity", enters a start time, end time, and description, clicks "Add" and yet no activity is added. I suspect this occurs when the activity would overlap with an existing activity
 * [x] similarly, activity names that begin with a `#` are now ignored. Attempting to add such an activity result in no activity being created.
 * [x] it is no longer possible to interact with the "timeline" view in the Add Activity window (#199)
 * [ ] category name completion seems not to work with capitalized category names (they simply don't appear in the completion list)

I'll try to chip away at this list in the coming weeks.

**Comment by ederag** [comment]
That was probably *not* the intention, and rather an implicit request for pointers, so it's OK,
but please let me mention that this post feels a bit obnoxious,
read from the maintainer side.

>  the collapsed "Overview" panel is significantly less useful than the previous Overview window as it is no longer possible to use the

The sentence is incomplete. Please file an issue with a clear description.
In general, separate issues are better for tracking,
and do not present such an accumulation effect.


Now the question I'm really curious about:

> the behavior of the "Add Activity" button is quite confusing, acting to either add a new activity or edit an existing activity depending upon whether there is an active selection

The [infotip](https://github.com/projecthamster/hamster/pull/454#issuecomment-538292973) proposed in #454 received zero support. 
As a "new"  user of the v2 interface, what it your opinion about this infotip ?


I'm taking a short break from hamster, so will let other answer in detail, here are stubs:
You might want to look at #423.

And add a `to_lower()` inside
https://github.com/projecthamster/hamster/blob/1ddd6bdcd4e07bae30a2b5a859e369ee7aa4bce3/src/hamster/widgets/activityentry.py#L628
(probably would accept a PR in this direction).

**Comment by GeraldJansen** [comment]
I agree with @ederag that separate issues would be better for cleaner discussion and progressive completion. Separate PR will be better for the same reason.

> * [ ]  the behavior of the "Add Activity" button is quite confusing, acting to either add a new activity or edit an existing activity depending upon whether there is an active selection

A new activity is also created when there is an active selection, but it is populated with the same activity name, category etc. as the selected activity. This is a very convenient way to resume a previous activity.

> * [ ]  attempting to add an activity sometimes fails inexplicably and silently; that is, one clicks "Add Activity", enters a start time, end time, and description, clicks "Add" and yet no activity is added. I suspect this occurs when the activity would overlap with an existing activity

A separate issue with reproducible steps would be good. We do need a notification method rather than silently ignoring invalid inputs. Also, a review of merging/splitting of overlapping activities is another open issue, but that has set aside for after 3.0.

> * [ ]  it is no longer possible to interact with the "timeline" view in the Add Activity window (#199)

This has already been mentioned/discussed in several other issues (eg. #313, but also more recently). 

Looking forward to your contributions!

**Comment by fantasai** [comment]
> A new activity is also created when there is an active selection, but it is populated with the same activity name, category etc. as the selected activity. This is a very convenient way to resume a previous activity.

This is nice, but the previous version could do that with just double-clicking on the activity, which was super convenient. (Editing the earlier activity required clicking on an edit button; but one edits less often than one picks up an earlier task, so this is appropriate.) Would love to have the same ergonomics in the new version.

**Comment by kaedroho** [comment]
I've been trying this out myself recently. Really nice work so far! Thanks for all your effort!

Just spotted a few minor issues:

 - It would be really nice if the "Add activity" window initialised to the selected date in the main UI
 - The "end" dropdown seems to be relative to the current time instead of the value in "start"
 - The `#` matching in tag seems to be more aggressive than before. I used to put tags at the beginning but doing that now makes it recognise the entire line as a tag instead of just the word it's prefixing

**Comment by ederag** [comment]
Thanks for the kind words, and for the report.

I should have closed this one (but there are interesting informations in it), 
as the title calls for reports bunching :wink:
Would you mind filing three separate issues ? That would be greatly appreciated.

*The problem is that when I'm ready to investigate a specific topic described here, 
I have to reread all, each time. This is overwhelming and inefficient.*

**Comment by ederag** [comment]
@kaedroho
> It would be really nice if the "Add activity" window initialised to the selected date in the main UI

Explained in a new separate issue: #544.

> The "end" dropdown seems to be relative to the current time instead of the value in "start"

Seems fixed (thanks for the report). Please file another issue with your use case otherwise.

> I used to put tags at the beginning of the entry

That's unexpected ! From the legacy code, it was never intended to be used like that.
It would be difficult to handle that while keeping other improvements and clarity.
But perhaps not impossible; we can discuss that in a separate issue if you wish.

**Comment by ederag** [comment]
@bgamari Thanks for the report, even if the form was not optimal.
Still interested in answers to questions in my [first comment](https://github.com/projecthamster/hamster/issues/471#issuecomment-550408814).

> attempting to add an activity sometimes fails inexplicably and silently;

Thanks, agreed, now tracked in [maintenance](https://github.com/projecthamster/hamster/projects/1#card-32830290).

> category name completion seems not to work with capitalized category names

It works here, except it is case sensitive. Ideally, that should be selectable, tracked in [maintenance](https://github.com/projecthamster/hamster/projects/1#card-32830418).
Please open a separate issue for further discussion on this subject.

Closing this issue before releasing the beta.

TLDR; for people reading the bottom line:
*not fully fixed, but please [file separate issues](https://github.com/projecthamster/hamster/issues/471#issuecomment-579950189)*
